nextcord/application_command.py

```diff
@@ -745,12 +745,50 @@ def from_callback(
                 skip_counter = 1
                 # TODO: use typing.get_type_hints when 3.9 is standard
                 typehints = typing_extensions.get_type_hints(self.callback, include_extras=True)
+                callback_params = signature(self.callback).parameters
+                possible_options: Set[str] = set()
+
+                for i, param in enumerate(callback_params.values()):
+                    if not (
+                        param.annotation is param.empty  # could be a self or interaction parameter
+                        or issubclass(
+                            param.annotation, Interaction
+                        )  # will always be an interaction parameter
+                        # TODO: use typing.Self when 3.11 is standard
+                        or param.annotation
+                        is typing_extensions.Self  # will always be a self parameter
+                        or isinstance(
+                            param.annotation, typing.TypeVar
+                        )  # will always be a self parameter
+                    ) and i in (0, 1):
+                        possible_options.add(param.name)
+
+                intersection = {param.name for param in callback_params.values()} - possible_options
```

Comment:
This variable should probably be renamed since the term "intersection" means the things in common, but this is more like the opposite since it is finding the set difference (in this case, the items that are *not* in common between the two sets). Therefore, I think `difference` would be a more appropriate variable name. Or perhaps something more descriptive.